src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerWithoutBinderBase.cs

```diff
@@ -70,7 +69,6 @@ public sealed override void Initialize(AnalysisContext context)
 
             Debug.Assert(!String.IsNullOrWhiteSpace(this.DeserializerTypeMetadataName));
             Debug.Assert(!String.IsNullOrWhiteSpace(this.SerializationBinderPropertyMetadataName));
-            Debug.Assert(cachedDeserializationMethodNames != null);
             Debug.Assert(!cachedDeserializationMethodNames.IsEmpty);
             Debug.Assert(this.BinderDefinitelyNotSetDescriptor != null);
             Debug.Assert(this.BinderMaybeNotSetDescriptor != null);
```

Comment:
If DoNotUseInsecureDeserializerMethodsBase.cs has its != null asserts removed, might as well remove these ones (or I can do it later).

Reply:
I only removed the ones that were leading to nullable warnings. I think you may want to take a pass over all the security analyzer files and audit all the existing asserts.